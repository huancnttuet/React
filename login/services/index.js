export { default as serverServices } from "./user";
