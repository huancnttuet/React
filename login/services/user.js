var user = require("../model/user.js");
var nodemailer = require("nodemailer");

module.exports = {
  signin: async (usernameSignIn, pwdSignIn) => {
    var valueSignIn = await user.checkSignIn(usernameSignIn, pwdSignIn);

    if (valueSignIn !== 0) {
      console.log("login true");
      return {
        code: "SUCCESS",
        message: "Đăng nhập thành công",
        result: {
          username: valueSignIn.username,
          id: valueSignIn.id
        }
      };
    } else {
      console.log("login false");
      return {
        code: "ERROR",
        message: "Sai mật khẩu"
      };
    }
  },
  signup: async (emailSignUp, usernameSignUp) => {
    const sendMail = pwd => {
      var transporter = nodemailer.createTransport({
        service: "Gmail",
        auth: {
          user: "huancnttuet@gmail.com",
          pass: "341997mok"
        }
      });

      var mailOptions = {
        from: "huancnttuet@gmail.com",
        to: "huancnttmta@gmail.com",
        subject: "Password demo",
        text: `Yourpassword: ${pwd}`
      };

      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          console.log(error);
        } else {
          console.log("Email sent: " + info.response);
        }
      });
    };
    const createPwdSignUp = () => {
      var length = 8,
        charset =
          "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789",
        retVal = "";
      for (var i = 0, n = charset.length; i < length; ++i) {
        retVal += charset.charAt(Math.floor(Math.random() * n));
      }
      return retVal;
    };

    var checkSU = await user.checkSignUp(emailSignUp, usernameSignUp);
    console.log(checkSU);
    if (!checkSU) {
      var pwdSignUp = createPwdSignUp();

      if (await user.createUser(emailSignUp, usernameSignUp, pwdSignUp)) {
        sendMail(pwdSignUp);
        return {
          code: "SUCCESS",
          message: "create new user success"
        };
      } else {
        return {
          code: "ERROR",
          message: "error create new user"
        };
      }
    } else {
      return {
        code: "EXIST",
        message: "user or mail exist"
      };
    }
  },
  changepwd: async (id, pwd, pwdNew) => {
    if (await user.checkIdPwd(id, pwd)) {
      if (await user.changePwd(id, pwdNew)) {
        return {
          code: "SUCCESS",
          message: "Đổi mật khẩu thành công"
        };
      } else {
        return {
          code: "ERROR",
          message: "Đã có lỗi xảy ra"
        };
      }
    } else {
      return {
        code: "SAIPASS",
        message: "Sai mật khẩu"
      };
    }
  },
  forgottenacc: async emailFA => {
    var pwd = await user.checkEmailFA(emailFA);
    const sendMail = pwd => {
      var transporter = nodemailer.createTransport({
        service: "Gmail",
        auth: {
          user: "huancnttuet@gmail.com",
          pass: "341997mok"
        }
      });

      var mailOptions = {
        from: "huancnttuet@gmail.com",
        to: "huancnttmta@gmail.com",
        subject: "Forgotten Password demo",
        text: `Yourpassword: ${pwd}`
      };

      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          console.log(error);
          return {
            code: "ERROR",
            message: "Lỗi trong quá trình gửi mail"
          };
        } else {
          console.log("Email sent: Quen pass " + info.response);
          return {
            code: "SUCCESS",
            message: "Pass đã được gửi lại về mail của bạn"
          };
        }
      });
    };
    if (pwd !== null) {
      sendMail(pwd);
    } else {
      return {
        code: "NOTFOUND",
        message: "Mail chưa được đăng ký"
      };
    }
  }
};
