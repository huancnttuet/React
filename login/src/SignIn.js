import React,{useState, useEffect} from 'react'
import {Form, Button, Col, Row, Container} from 'react-bootstrap'
import axios from 'axios'

export default function SignIn(props) {
  const username = useFormInput('')
  const pwd = useFormInput('')
  const [stateLogin , setStateLogin] = useState(false)
  console.log(props);
  function handleClick() {
    var data = {
      usernameSignIn: username.value,
      pwdSignIn: pwd.value
    }

    axios.post('http://localhost:8000/login', {data}).then((res) => {
      console.log(res.data.login);
      if(res.data.login){
        setStateLogin(true);
      }
    })
  }
  if(stateLogin == true){
    return (
      <div>
        <h1>CHÀO MỪNG BẠN =))</h1>
      </div>
    )
  }
  else {
    return(
      <Container>
        <Row>
          <Col>
            <Form>
              <Form.Group controlId="formBasicUsername" {...username}>
                <Form.Label>Username</Form.Label>
                <Form.Control type="text" placeholder="Enter username" />
                <Form.Text className="text-muted">
                  We'll never share your username with anyone else.
                </Form.Text>
              </Form.Group>

              <Form.Group controlId="formBasicPassword" {...pwd}>
                <Form.Label>Password</Form.Label>
                <Form.Control type="password" placeholder="Password" />
              </Form.Group>
              <Form.Group controlId="formBasicChecbox">
                <Form.Check type="checkbox" label="Check me out" />
              </Form.Group>
              <Button variant="primary"  onClick={handleClick}>
                Submit
              </Button>
            </Form>
          </Col>
        </Row>
      </Container>
    )
  }

}

function useFormInput(initial){
  const [value, setValue] = useState(initial)

  function handleChange(e){
    setValue(e.target.value)
  }

  return {
    value,
    onChange: handleChange
  }

}
