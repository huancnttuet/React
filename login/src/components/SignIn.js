import React,{useState } from 'react'
import {Form, Button, Col, Row, Container} from 'react-bootstrap'
import axios from 'axios'
import {useGlobal} from 'reactn'



function SignIn(props) {
  const [global, setGlobal] = useGlobal()
  const username = useFormInput('')
  const pwd = useFormInput('')
  const [message, setMessage] = useState('')

  console.log(global);
  function handleClick() {
    var data = {
      usernameSignIn: username.value,
      pwdSignIn: pwd.value
    }
    axios.post('http://localhost:8000/signin', {data}).then((res) => {
      console.log(res.data.login);
      if(res.data.login){
          global.dispatch({
            type: 'LOGIN',
            payload: res.data.id
          })
      } else {
        setMessage(res.data.message)
      }
    })
  }
  console.log(global.state.authenticate);
  if(global.state.authenticate === true){
    return (
      <div>

        <h1>CHÀO MỪNG BẠN </h1>
      </div>
    )
  }
  else {
    return(
      <div>

        <Container>
          <Row style={{marginTop:50}}>
            <Col>
              {props.authenticate}
            </Col>
            <Col >
              <Form>
                <Form.Group controlId="form-basic-username" {...username}>
                  <Form.Label>Username</Form.Label>
                  <Form.Control type="text" placeholder="Enter username" />
                  <Form.Text className="text-muted">
                    We'll never share your username with anyone else.
                  </Form.Text>
                </Form.Group>

                <Form.Group controlId="formBasicPassword" {...pwd}>
                  <Form.Label>Password</Form.Label>
                  <Form.Control type="password" placeholder="Password" />
                </Form.Group>
                <Form.Group controlId="formBasicChecbox">
                  <Form.Check type="checkbox" label="Check me out" />
                </Form.Group>
                <Button variant="primary" id='login-btn' onClick={handleClick}>
                  Submit
                </Button>
                <p>{message}</p>
                <a href='http://localhost:3000/forgottenacc'>Forgotten account?</a>
              </Form>
            </Col>
            <Col>
            </Col>
          </Row>
        </Container>

      </div>
    )
  }

}

function useFormInput(initial){
  const [value, setValue] = useState(initial)

  function handleChange(e){
    setValue(e.target.value)
  }

  return {
    value,
    onChange: handleChange
  }
}



export default SignIn
