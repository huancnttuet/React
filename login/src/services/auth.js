import axios from 'axios'

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/signin`, params);
}

const signup = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/signup`, params);
}

const changepwd = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changepwd`, params);
}

const forgottenacc = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgottenacc`, params);
}

export default {
  login,
  signup,
  changepwd,
  forgottenacc,
}
