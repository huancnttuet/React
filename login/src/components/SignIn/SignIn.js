import React, { useState, useGlobal } from "reactn";
import { Form, Button, Col, Row, Container } from "react-bootstrap";
import { authServices } from "services";

function SignIn(props) {
  const [authenticate, setAuthenticate] = useGlobal("authenticate");
  const [id, setId] = useGlobal("id");
  const username = useFormInput("");
  const pwd = useFormInput("");
  const [message, setMessage] = useState("");

  function handleClick() {
    var data = {
      usernameSignIn: username.value,
      pwdSignIn: pwd.value
    };
    authServices.login(data).then(res => {
      setMessage(res.data.message);
      if (res.data.code === "SUCCESS") {
        setAuthenticate(true);
        setId(res.data.result.id);
      } else {
      }
    });
  }

  const goToForgottenAccount = () => {
    props.history.push("/forgottenacc");
  };

  if (authenticate === true) {
    return (
      <>
        <h1>CHÀO MỪNG BẠN </h1>
      </>
    );
  } else {
    return (
      <div>
        <Container>
          <Row style={{ marginTop: 50 }}>
            <Col />
            <Col>
              <Form>
                <Form.Group controlId="username-signin" {...username}>
                  <Form.Label>Username</Form.Label>
                  <Form.Control type="text" placeholder="Enter username" />
                  <Form.Text className="text-muted">
                    We'll never share your username with anyone else.
                  </Form.Text>
                </Form.Group>

                <Form.Group controlId="password-signin" {...pwd}>
                  <Form.Label>Password</Form.Label>
                  <Form.Control type="password" placeholder="Password" />
                </Form.Group>
                <Form.Group controlId="formBasicChecbox">
                  <Form.Check type="checkbox" label="Check me out" />
                </Form.Group>
                <Button variant="primary" id="login-btn" onClick={handleClick}>
                  Submit
                </Button>
                <p>{message}</p>
                <div
                  style={{ cursor: "pointer" }}
                  onClick={goToForgottenAccount}
                >
                  Forgotten account?
                </div>
              </Form>
            </Col>
            <Col />
          </Row>
        </Container>
      </div>
    );
  }
}

function useFormInput(initial) {
  const [value, setValue] = useState(initial);

  function handleChange(e) {
    setValue(e.target.value);
  }

  return {
    value,
    onChange: handleChange
  };
}

export default SignIn;
