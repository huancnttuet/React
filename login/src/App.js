import React from 'react';
import './css/App.css';

import TopPage from './TopPage'
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Home from './Home'
import SignIn from './SignIn'
import SignUp from './SignUp'

function App() {
  return (
    <Router>
       <div>
         <TopPage />

         <div >
           <Route exact path="/" component={Home} />
           <Route path="/signin/:id" component={SignIn} />
           <Route path="/signup" component={SignUp} />
         </div>
       </div>


     </Router>
  );
}

export default App;
