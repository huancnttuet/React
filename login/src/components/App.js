import React, { useReducer, setGlobal, useGlobal } from "reactn";
import "./css/App.css";

import { BrowserRouter as Router, Route } from "react-router-dom";
import Home from "./Home";
import SignIn from "./SignIn";
import SignUp from "./SignUp";
import ChangePwd from "./ChangePwd";
import ForgottenAcc from "./ForgottenAcc";

import TopPage from "./TopPage";

// const reducer = (state, action) => {
//   switch (action.type) {
//     case "LOGIN":
//       localStorage.setItem("authenticate", true);
//       localStorage.setItem("id", action.payload);
//       return { authenticate: true, id: action.payload };
//     case "LOGOUT":
//       localStorage.setItem("authenticate", false);
//       localStorage.setItem("id", 0);
//       return { authenticate: false, id: 0 };
//     default:
//       return state;
//   }
// };

setGlobal({
  authenticate: false,
  id: 0
});

function App(props) {
  // const [state, dispatch] = useReducer(reducer, {
  //   authenticate: localStorage.getItem("authenticate"),
  //   id: localStorage.getItem("id")
  // });

  const [id] = useGlobal("id");

  // var pathHome;
  // if (id === 0) {
  //   pathHome = "/";
  // } else {
  //   pathHome = `/${id}`;
  // }

  console.log(id);

  return (
    <Router>
      <TopPage />
      <div>
        {/* <Route exact path={pathHome} component={Home} /> */}
        <Route exact path="/" component={props => <SignIn {...props} />} />
        <Route path="/signup" component={SignUp} />
        <Route path="/changepwd" component={ChangePwd} />
        <Route path="/forgottenacc" component={ForgottenAcc} />
      </div>
    </Router>
  );
}

export default App;
