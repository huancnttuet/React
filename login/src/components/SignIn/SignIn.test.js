import React from "react";
import { shallow, mount } from "enzyme";
import App from "../App";
import SignIn from "./SignIn";
import { authServices } from "services";

import renderer from "react-test-renderer";

let wrapper;
beforeEach(() => {
  wrapper = mount(<App />);
});

afterEach(() => {
  wrapper.unmount();
});

describe("SignIn", () => {
  it("click button login", () => {
    const loginMock = jest.spyOn(authServices, "login");
    const wrapper1 = mount(<SignIn />);
    const login = renderer.create(<SignIn />).toJSON();
    expect(login).toMatchSnapshot();
    wrapper1.find("#username-signin").simulate("change", {
      target: {
        value: "huanhuan"
      }
    });
    wrapper1.find("#password-signin").simulate("change", {
      target: {
        value: "12"
      }
    });
    wrapper1
      .find("#login-btn")
      .at(0)
      .simulate("click");
    wrapper1
      .find("#login-btn")
      .at(1)
      .simulate("click");
    expect(loginMock).toHaveBeenCalled();
  });

  test("test API", () => {
    const data = {
      usernameSignIn: `huanhuan`,
      pwdSignIn: `1`
    };
    return authServices.login(data).then(data => {
      expect(data.data).toMatchObject({ code: "SUCCESS" });
    });
  });
});
