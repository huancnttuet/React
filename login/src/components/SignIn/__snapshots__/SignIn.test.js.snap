// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn click button login 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "marginTop": 50,
        }
      }
    >
      <div
        className="col"
      />
      <div
        className="col"
      >
        <form
          className=""
        >
          <div
            className="form-group"
            onChange={[Function]}
            value=""
          >
            <label
              className="form-label"
              htmlFor="username-signin"
            >
              Username
            </label>
            <input
              className="form-control"
              id="username-signin"
              placeholder="Enter username"
              type="text"
            />
            <small
              className="text-muted form-text"
            >
              We'll never share your username with anyone else.
            </small>
          </div>
          <div
            className="form-group"
            onChange={[Function]}
            value=""
          >
            <label
              className="form-label"
              htmlFor="password-signin"
            >
              Password
            </label>
            <input
              className="form-control"
              id="password-signin"
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="form-group"
          >
            <div
              className="form-check"
            >
              <input
                className="form-check-input"
                disabled={false}
                id="formBasicChecbox"
                type="checkbox"
              />
              <label
                className="form-check-label"
                htmlFor="formBasicChecbox"
                title=""
                type="checkbox"
              >
                Check me out
              </label>
            </div>
          </div>
          <button
            className="btn btn-primary"
            disabled={false}
            id="login-btn"
            onClick={[Function]}
            type="button"
          >
            Submit
          </button>
          <p>
            
          </p>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            Forgotten account?
          </div>
        </form>
      </div>
      <div
        className="col"
      />
    </div>
  </div>
</div>
`;
